src/Development/Rattle/Server.hs

```diff
@@ -262,7 +267,15 @@ cmdRattleRun rattle@Rattle{..} cmd@(Cmd _ opts args) startTimestamp hist msgs =
                     cmdRattleFinished rattle startTimestamp cmd t False
                 Nothing -> do
                     start <- timer
-                    (opts2, c) <- display [] $ cmdRattleRaw ui opts args
+                    -- close file then reopen
+                    (opts2, c) <- if isJust $ rattleOut options
+                                  then modifyVar outFile (\fh -> do
+                                                             hClose fh
+                                                             p <- display [] $ cmdRattleRaw ui opts args
```

Comment:
Why do we close the file then reopen here?

Reply:
The file was specified as where stdout and stderr of cmds should go, so if we don't close the file then 'cmd' cannot open and write to it.